src/components/ProductFilterSidebar.js

```diff
@@ -0,0 +1,142 @@
+import { useFilter } from "../contexts/FilterContext";
+import { CategoriesData } from "../data";
+
+export const ProductFilterSidebar = ({ showFilterBar }) => {
+  const { filter, dispatchFilter } = useFilter();
+  const { sortBy, includeOutOfStock, maxPriceRange, categories, ratingRange } = filter;
+  return (
+    <>
+      <div className={`side-bar product-sidebar box-shd w-20p ${showFilterBar ? "show-filter" : ""}`}>
+        <div className="side-bar-title title row-flex p-v-2 p-h-3">
+          <p>Filters</p>
+          <p className="clear-all cursor" onClick={() => dispatchFilter({ type: "CLEARALL" })}>
+            Clear all
+          </p>
+          <p className="icon-toggle icon-btn sidebar-close rd-bdr grid-ctr colored-text">
+            <i className="fa fa-close" aria-hidden="true"></i>
+          </p>
+          <div className="line-decoration"></div>
+        </div>
+        <ul className="sidebar-items no-bullet col-flex flex-start no-wrap p-h-2 m-b-3">
+          <li className="filter-list p-l-2 m-v-1">CATEGORIES</li>
+          {CategoriesData?.map(({ id, name }) => {
```

Comment:
Doubt: this is new to me, why is ```?``` needed before the map here? What happens without it, I mean it should work right?

Reply:
yes, technically it should. But I've noticed while getting data from API, if the data is `null` then chaining it to the map function gives an error. So for the safer side, I've added optional chaining operator (?.) to eliminate that error.